_posts/aws/2019-07-16-AWS-CloudFormation-HowTo-enable.md

```diff
@@ -0,0 +1,131 @@
+type: Document
+---
+
+## Background
+
+This document will show you how to enable the CloudFormation stage.  This document also assumes that you have configured an AWS account.  If you need more instructions on configuring your AWS account please refer to either [Deploying to AWS from Spinnaker (using IAM instance roles)](https://docs.armory.io/spinnaker-install-admin-guides/add-aws-account-iam/) or [Deploying to AWS from Spinnaker (using IAM credentials)](https://docs.armory.io/spinnaker-install-admin-guides/add-aws-account/) 
+
+## Preconditions
+&nbsp;
+### Enable Cloudformation in Spinnaker
+
+First, we need to enable CloudDriver by adding a `clouddriver-local.yml` file to your hal config profiles directory e.g. `.hal/default/profiles/clouddriver-local.yml`.
+```yaml
+aws:
+  features:
+    cloudFormation:
+      enabled: true
+```
+You can check your configuration by running `hal deploy apply`.
+
+&nbsp;
+### Halconfig 
+
+Make sure to enable the aws provider and have set at least one region
+
+```yaml
+aws:
+  enabled: true
+  accounts:
+  - name: aws-1
+    requiredGroupMembership: []
+    providerVersion: V1
+    permissions: {}
+    accountId: '123456789123'
+    regions:
+    - name: us-east-1
+    - name: us-west-2
+    assumeRole: role/SpinnakerManagedRole
+  primaryAccount: aws-1
+  bakeryDefaults:
+    templateFile: aws-ebs-shared.json
+    baseImages: []
+    awsAssociatePublicIpAddress: true
+    defaultVirtualizationType: hvm
+  defaultKeyPairTemplate: '-keypair'
+  defaultRegions:
+  - name: us-west-2
+  defaults:
+    iamRole: BaseIAMRole
+```
+
+&nbsp;
+### CloudFormation permissions in Policy
+
+Another thing to check is the `cloudformation:*` permission in the *"Action"* section inside your **Policy**, like in the following example. 
+
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": "sts:AssumeRole",
+            "Resource": "arn:aws:iam::123456789123:role/SpinnakerManagedRole"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:*",
+                "cloudformation:*"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+_for more details please refer to [Creating a Managing Account IAM Policy in your primary AWS Account](https://docs.armory.io/spinnaker-install-admin-guides/add-aws-account/#iam-user-part-3-creating-a-managing-account-iam-policy-in-your-primary-aws-account)_
+
+## Creating your Pipeline
+
+After you've deployed your changes, you should now be able to see the new stages when configuring your pipeline. 
+Simply select the stage, and provide the values: example below: 
+
+![](/images/cloudformation-pipeline1.png)
+
+In this exaple we are going to deploy a simple s3 bucket with the follow template, (Copy and paste in the **Text Source** section)
+
```

Comment:
Add a comment about the name having to be different for the cloud formation template

Reply:
Note added, for the "Stack name" should be unique.